packages-exp/auth-exp/src/core/user/additional_user_info.ts

```diff
@@ -44,8 +44,8 @@ export function _fromIdTokenResponse(
     ];
     if (signInProvider) {
       const filteredProviderId =
-        providerId !== externs.ProviderId.ANONYMOUS &&
-        providerId !== externs.ProviderId.CUSTOM
+        signInProvider !== externs.ProviderId.ANONYMOUS &&
+        signInProvider !== externs.ProviderId.CUSTOM
```

Comment:
this wasn't caught by any tests?

Reply:
No; the assumptions for anonymous and custom in the test were wrong and weren't caught in this logic branch. Updated.